tests/unit/dash/development/test_base_component.py

```diff
@@ -469,40 +464,37 @@ def test_iter(self):
                 Component(children=[Component(id='8')]),
             ]
         )
-        # test keys()
-        keys = [k for k in list(c.keys())]
-        self.assertEqual(keys, ['2', '3', '4', '5', '6', '7', '8'])
-        self.assertEqual([i for i in c], keys)
 
-        # test values()
-        components = [i for i in list(c.values())]
-        self.assertEqual(components, [c[k] for k in keys])
+        mixins = ['clear', 'get', 'items', 'keys', 'pop', 'popitem',
+                  'setdefault', 'update', 'values']
+
+        for m in mixins:
+            self.assertFalse(hasattr(c, m), m)
+
+        keys = ['2', '3', '4', '5', '6', '7', '8']
 
-        # test __iter__()
         for k in keys:
             # test __contains__()
             self.assertTrue(k in c)
+            # test __getitem__()
+            self.assertEqual(c[k].id, k)
```

Comment:
since we already switched to pytest now, I think it's better to use native `assert`  like 

assert c[k].id == k, "brief description on the criteria"